src/main/scala/com/datastax/sparkstress/ReadTask.scala

```diff
@@ -142,8 +220,21 @@ class JWCAllColumns(config: Config, sc: SparkContext) extends ReadTask(config, s
 }
 
 /**
+  * SparkSQL join table with itself
+  */
+class SparkSqlJoinAllColumns(config: Config, sc: SparkContext) extends ReadTask(config, sc) {
+  def run(): Unit = {
+    val count = sqlContext.sql("SELECT COUNT(table_b.store) FROM ks.tab AS table_b JOIN ks.tab AS table_a ON table_b.store = table_a.store")
+      .toDF("count")
+      .first.getAs[Long]("count")
```

Comment:
Because we're result is not raw rows, but a count column, so we need this little extra bit to get the value, otherwise we get just 1.